src/code_generator.cpp

```diff
@@ -242,13 +242,25 @@ bool generate_type_alias(code_generator& generator, const cpp_type_alias& alias,
     code_generator::output output(type_safe::ref(generator), type_safe::ref(alias), cur_access);
     if (output)
     {
-        output << keyword("using") << whitespace << identifier(alias.name()) << operator_ws
-               << punctuation("=") << operator_ws;
-        if (output.options() & code_generator::exclude_target)
-            output.excluded(alias);
+        if (alias.use_c_style())
+        {
+            output << keyword("typedef") << whitespace;
+            if (output.options() & code_generator::exclude_target)
+                output.excluded(alias);
+            else
+                detail::write_type(output, alias.underlying_type(), "");
+            output << whitespace << identifier(alias.name()) << punctuation(";") << newl;
```

Comment:
That doesn't generate the correct syntax for complex types `typedef int (*)(int) foo` vs `typedef int (*foo)(int)`. The third parameter of `write_type` is used for that, you can specify the name there and it will be printed at the correct position.

Reply:
Ok, makes sense. I hadn't really considered the syntax differences with complex types. It should be fixed now